From 31734bd5b22a5025f15c8a1618f37862034f9b6a Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Fri, 18 Oct 2024 19:30:31 +0530
Subject: [PATCH] For Testing thermal commenting failsafe mode as it is causing
 issue need to debug more on it ,

jira link: https://oneami.atlassian.net/browse/OT-7883

Tested:

Stepwise configuration:
{
                "Class": "temp",
                "Inputs": [
                    "EVB_Temp"
                ],
                "Name": "EVB_Temp",
                "NegativeHysteresis": 0,
                "Output": [
                    10.0,
                    30.0,
                    50.0,
                    70.0
                ],
                "PositiveHysteresis": 0,
                "Reading": [
                    18.0,
                    20.0,
                    22.0,
                    35.0
                ],
                "Type": "Stepwise",
                "Zones": [
                    "Zone 1"
                ]
            },

root@evb-npcm845:/sys/class/hwmon/hwmon1# ipmitool sensor list
Pwm 1            | 49.784     | unspecified | ok    | na        | na        | na        | na        | na        | na

Pwm 2            | 49.784     | unspecified | ok    | na        | na        | na        | na        | na        | na

Pwm 3            | 49.784     | unspecified | ok    | na        | na        | na        | na        | na        | na

Pwm 4            | 99.960     | unspecified | ok    | na        | na        | na        | na        | na        | na

Fan1             | 0.000      | RPM        | ok    | na        | 495.000   | na        | 8525.000  | 11495.000 | na

Fan2             | 0.000      | RPM        | ok    | na        | 495.000   | na        | 8525.000  | 11495.000 | na

Fan3             | 14025.000  | RPM        | cr    | na        | 495.000   | na        | 8525.000  | 11495.000 | na

Fan4             | 0.000      | RPM        | ok    | na        | 495.000   | na        | 8525.000  | 11495.000 | na

EVB Temp         | 29.000     | degrees C  | ok    | na        | 0.000     | 15.000    | 50.000    | 65.000    | na

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 pid/fancontroller.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/pid/fancontroller.cpp b/pid/fancontroller.cpp
index 378ef9b..ffe6aa2 100644
--- a/pid/fancontroller.cpp
+++ b/pid/fancontroller.cpp
@@ -152,10 +152,11 @@ void FanController::outputProc(double value)
             // The computed PWM is still allowed to rise higher than
             // failsafe PWM if STRICT_FAILSAFE_PWM is NOT defined.
             // This is the default behavior.
-            if (percent < failsafePercent)
+            /*if (percent < failsafePercent)
             {
                 percent = failsafePercent;
-            }
+            }*/
+	    std::cerr << "failsafePercent " << failsafePercent << "\n";
 #endif
         }
 
-- 
2.25.1

